master/buildbot/test/e2e/test_master_worker_create_start_stop.py

```diff
@@ -0,0 +1,921 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import print_function
+
+import os
+import re
+import socket
+import textwrap
+
+from fastjsonrpc.client import Proxy
+from fastjsonrpc.client import jsonrpc
+from twisted.internet import reactor as global_reactor
+from twisted.internet import defer
+from twisted.internet import task
+from twisted.internet import utils
+from twisted.python import log
+from twisted.trial import unittest
+from txrequests import Session
+
+from buildbot.test.util import dirs
+from buildbot.test.util.decorators import skipUnlessInstalled
+from buildbot.test.util.decorators import skipUnlessPlatformIs
+
+try:
+    from shutil import which
+except ImportError:
+    # Backport of shutil.which() from Python 3.3.
+    from shutilwhich import which
+
+try:
+    from textwrap import indent
+except ImportError:
+    # textwrap.indent() implementation from Python 3.3
+    def indent(text, prefix, predicate=None):
+        """Adds 'prefix' to the beginning of selected lines in 'text'.
+
+        If 'predicate' is provided, 'prefix' will only be added to the lines
+        where 'predicate(line)' is True. If 'predicate' is not provided,
+        it will default to adding 'prefix' to all non-empty lines that do not
+        consist solely of whitespace characters.
+        """
+        if predicate is None:
+            def predicate(line):  # pylint: disable=function-redefined
+                return line.strip()
+
+        def prefixed_lines():
+            for line in text.splitlines(True):
+                yield (prefix + line if predicate(line) else line)
+        return ''.join(prefixed_lines())
+
+
+def get_open_port():
+    # TODO: This is synchronous code which might be blocking, which is
+    # unacceptable in Twisted.
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(('', 0))
+    s.listen(1)
+    port = s.getsockname()[1]
+    s.close()
+    return port
+
+
+SHELL_COMMAND_TEST_CONFIG = """\
+from buildbot.plugins import *
+
+c = BuildmasterConfig = {}
+
+c['workers'] = [worker.Worker('example-worker', 'pass')]
+c['protocols'] = {'pb': {'port': 9989}}
+c['schedulers'] = [
+    schedulers.ForceScheduler(
+        name='force',
+        builderNames=['runtests'])
+]
+
+factory = util.BuildFactory(
+    [steps.ShellCommand(command=['echo', "Test echo"])])
+
+c['builders'] = [
+    util.BuilderConfig(name='runtests', workernames=['example-worker'],
+        factory=factory),
+]
+
+c['buildbotURL'] = 'http://localhost:8010/'
+c['www'] = dict(port=8010, plugins={})
+c['db'] = {
+    'db_url' : 'sqlite:///state.sqlite',
+}
```

Comment:
Please dont forget to disable BuildbotNetUsageData on those tests. I can see a spike of 150 installs on events.buildbot.net :)